src/sentry/api/invite_helper.py

```diff
@@ -33,16 +34,34 @@ def remove_invite_details_from_session(request: Request) -> None:
     request.session.pop("invite_token", None)
 
 
-def get_invite_details(request: Request) -> Tuple[str, int]:
-    """Returns tuple of (token, member_id) from request session"""
-    return request.session.get("invite_token", None), request.session.get("invite_member_id", None)
+@dataclass(frozen=True, eq=True)
+class InviteDetail:
+    token: str | None
+    member_id: int | None
+    user_id: int | None
+    is_authenticated: bool
+    is_verified: bool
+
+
+def get_invite_details(request: Request) -> InviteDetail:
+    """Extracts invite details from request session"""
+    user = request.user
+    primary_email = UserEmail.objects.get_primary_email(user) if isinstance(user, User) else None
```

Comment:
Longer term we might need to include the primary email address on `APIUser` as doing additional RPC calls for emails seems expensive.